media-service/src/main/java/org/arpha/service/BlobServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package org.arpha.service;
+
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.implementation.BlobsImpl;
+import com.azure.storage.blob.sas.BlobSasPermission;
+import com.azure.storage.blob.sas.BlobServiceSasSignatureValues;
+import com.azure.storage.common.sas.SasProtocol;
+import lombok.AllArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.tuple.Pair;
+import org.arpha.dto.media.enums.AccessType;
+import org.arpha.exception.GenerateFileLinkException;
+import org.arpha.property.AzureStorageProperties;
+import org.arpha.utils.Boxed;
+import org.springframework.stereotype.Service;
+
+import java.time.OffsetDateTime;
+@Service
+@RequiredArgsConstructor
+public class BlobServiceImpl implements BlobService {
+
+    private final AzureStorageProperties azureStorageProperties;
+    private final BlobContainerClient blobContainerClient;
+
+    @Override
+    public String generateLink(String fileName, AccessType accessType) {
+        return Boxed
+                .of(accessType)
+                .mapToBoxed(accessType1 -> BlobSasPermission.parse(accessType1.getPermission()))
+                .mapToBoxed(blobSasPermission -> new BlobServiceSasSignatureValues(OffsetDateTime.now().plusSeconds(
+                        azureStorageProperties.linkExpiration()), blobSasPermission))
+                .mapToBoxed(blobSignature -> blobSignature.setProtocol(SasProtocol.HTTPS_ONLY))
+                .mapToBoxed(blobSignature -> blobSignature.setCacheControl("no-cache"))
```

Comment:
Can you please move this piece of code to another method? It looks quite like mess if you do this in a single chain

Reply:
If it was at least 3 chains with set I'd agree, but creating a new method only for 2 lines seems like redundant thing. As fat as I concerned the best methods are considered to be from 5 to 10 lines of code,  my method contains 10 lines of logic so no need create separated method.